src/Internal/Payment/PaymentMethod/SavedPaymentMethod.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Class SavedPaymentMethod
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\PaymentMethod;
+
+use WC_Payment_Token;
+
+/**
+ * Representation of a newly entered payment method.
+ */
+class SavedPaymentMethod implements PaymentMethodInterface {
+	/**
+	 * Contains the WooCommerce token.
+	 *
+	 * @var WC_Payment_Token
+	 */
+	private $token;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param WC_Payment_Token $token The WooCommerce token.
+	 */
+	public function __construct( WC_Payment_Token $token ) {
+		$this->token = $token;
+	}
+
+	/**
+	 * Returns the ID of the payment method.
+	 *
+	 * @return string
+	 */
+	public function get_id(): string {
+		return $this->token->get_token();
+	}
+
+	/**
+	 * Retrieves the data of the payment method for storage.
+	 *
+	 * @return array
+	 */
+	public function get_data(): array {
+		return [
+			'type' => 'saved',
+			'id'   => $this->get_id(),
```

Comment:
I don't think `$this->get_id()` is suitable here. To construct the object from this data, we'd need the token ID.
```suggestion
			'id'   => $this->token->get_id(),
```

Reply:
Good point. Addressed https://github.com/Automattic/woocommerce-payments/pull/7020/commits/b9d4a92463953a5e494e9800c0642520248363ef